scene/gui/tab_bar.cpp

```diff
@@ -324,15 +326,30 @@ void TabBar::gui_input(const Ref<InputEvent> &p_event) {
 }
 
 void TabBar::_shape(int p_tab) {
-	tabs.write[p_tab].text_buf->clear();
-	tabs.write[p_tab].text_buf->set_width(-1);
-	if (tabs[p_tab].text_direction == Control::TEXT_DIRECTION_INHERITED) {
-		tabs.write[p_tab].text_buf->set_direction(is_layout_rtl() ? TextServer::DIRECTION_RTL : TextServer::DIRECTION_LTR);
+	TabBar::Tab *tab = &tabs.write[p_tab];
+	tab->text_buf->clear();
+	tab->text_buf->set_width(-1);
+	if (tab->text_direction == Control::TEXT_DIRECTION_INHERITED) {
+		tab->text_buf->set_direction(is_layout_rtl() ? TextServer::DIRECTION_RTL : TextServer::DIRECTION_LTR);
 	} else {
-		tabs.write[p_tab].text_buf->set_direction((TextServer::Direction)tabs[p_tab].text_direction);
+		tab->text_buf->set_direction((TextServer::Direction)tabs[p_tab].text_direction);
 	}
-
-	tabs.write[p_tab].text_buf->add_string(atr(tabs[p_tab].text), theme_cache.font, theme_cache.font_size, tabs[p_tab].language);
+	// italic support
+	Ref<Font> requested_font = theme_cache.font;
+	if (tab->title_style == ITALIC) {
+		Ref<FontVariation> italic_font;
+		italic_font.instantiate();
+		italic_font->set_base_font(requested_font);
+		italic_font->set_variation_transform(Transform2D(1.0, 0.2, 0.0, 1.0, 0.0, 0.0)); // TODO: There's room for performance improvement here
+		requested_font = italic_font;
+	}
+	// Fallback
+	if (!requested_font.is_valid()) {
+		ERR_PRINT_ONCE("Invalid italic font, using default font");
+		requested_font = theme_cache.font;
+	}
+	// Calculate font size
+	tab->text_buf->add_string(atr(tabs[p_tab].text), requested_font, theme_cache.font_size, tabs[p_tab].language);
```

Comment:
This is something that should be done with existing italic font in the theme and with the help of theme items.

```suggestion
	if (tab->title_style == ITALIC) {
		tab->text_buf->add_string(atr(tabs[p_tab].text), theme_cache.italics_font, theme_cache.italics_font_size, tabs[p_tab].language);
	} else {
		tab->text_buf->add_string(atr(tabs[p_tab].text), theme_cache.font, theme_cache.font_size, tabs[p_tab].language);
	}
```

Theme items definitions:
```diff
diff --git a/editor/themes/editor_fonts.cpp b/editor/themes/editor_fonts.cpp
index c13ee6e6b0..b5ff174ab3 100644
--- a/editor/themes/editor_fonts.cpp
+++ b/editor/themes/editor_fonts.cpp
@@ -421,6 +421,8 @@ void editor_register_fonts(const Ref<Theme> &p_theme) {
 	p_theme->set_font("italics_font", "RichTextLabel", italic_fc);
 	p_theme->set_font("bold_italics_font", "RichTextLabel", bold_italic_fc);
 
+	p_theme->set_font("italics_font", "TabBar", italic_fc);
+
 	// Documentation fonts
 	p_theme->set_font_size("doc_size", EditorStringName(EditorFonts), int(EDITOR_GET("text_editor/help/help_font_size")) * EDSCALE);
 	p_theme->set_font("doc", EditorStringName(EditorFonts), default_fc);
diff --git a/scene/gui/tab_bar.cpp b/scene/gui/tab_bar.cpp
index d20fef8164..49383f308e 100644
--- a/scene/gui/tab_bar.cpp
+++ b/scene/gui/tab_bar.cpp
@@ -1887,6 +1887,9 @@ void TabBar::_bind_methods() {
 	BIND_THEME_ITEM(Theme::DATA_TYPE_FONT_SIZE, TabBar, font_size);
 	BIND_THEME_ITEM(Theme::DATA_TYPE_CONSTANT, TabBar, outline_size);
 
+	BIND_THEME_ITEM(Theme::DATA_TYPE_FONT, TabBar, italics_font);
+	BIND_THEME_ITEM(Theme::DATA_TYPE_FONT_SIZE, TabBar, italics_font_size);
+
 	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_ICON, TabBar, close_icon, "close");
 	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_STYLEBOX, TabBar, button_pressed_style, "button_pressed");
 	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_STYLEBOX, TabBar, button_hl_style, "button_highlight");
diff --git a/scene/theme/default_theme.cpp b/scene/theme/default_theme.cpp
index fa83e06315..a27084fa03 100644
--- a/scene/theme/default_theme.cpp
+++ b/scene/theme/default_theme.cpp
@@ -918,6 +918,9 @@ void fill_default_theme(Ref<Theme> &theme, const Ref<Font> &default_font, const
 	theme->set_font("font", "TabBar", Ref<Font>());
 	theme->set_font_size("font_size", "TabBar", -1);
 
+	theme->set_font("italics_font", "TabBar", italics_font);
+	theme->set_font_size("italics_font_size", "TabBar", -1);
+
 	theme->set_color("font_selected_color", "TabBar", control_font_hover_color);
 	theme->set_color("font_hovered_color", "TabBar", control_font_hover_color);
 	theme->set_color("font_unselected_color", "TabBar", control_font_low_color);
```

Reply:
Missed header part:

```diff
diff --git a/scene/gui/tab_bar.h b/scene/gui/tab_bar.h
index 65a1d5bd4f..02d0d3b20c 100644
--- a/scene/gui/tab_bar.h
+++ b/scene/gui/tab_bar.h
@@ -136,6 +136,9 @@ private:
 		int font_size;
 		int outline_size = 0;
 
+		Ref<Font> italics_font;
+		int italics_font_size;
+
 		Color font_selected_color;
 		Color font_hovered_color;
 		Color font_unselected_color;
```